src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutBase.cs

```diff
@@ -96,6 +96,13 @@ private void OnPopupOpened(object sender, object e)
 		{
 			if (_popup.Child is FrameworkElement child)
 			{
+#if __IOS__
+				// workaround for #2984
+				var availableSize = ApplicationView.GetForCurrentView().VisibleBounds.Size;
+				child.MaxHeight = availableSize.Height;
```

Comment:
I don't feel like this is the right approach. However, I was unable to trace where this value is set.

Reply:
Here are some note that I took:
```
><Uno_UI_Controls_Window: 0x7ffb93f11d20; baseClass = UIWindow; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x6000004cc210>; layer = <UIWindowLayer: 0x600000ff3560>>-(414x896)@(0,0) d:<native/unk> Visible
  ><UITransitionView: 0x7ffb959a4f60; frame = (0 0; 414 896); autoresize = W+H; layer = <CALayer: 0x600000141120>>-(414x896)@(0,0) d:<native/unk> Visible
  >Windows.UI.Xaml.Controls.FlyoutBasePopupPanel-(414x896)@(0,0) d:414,896 Visible
    >Windows.UI.Xaml.Controls.FlyoutPresenter-(414x758)@(0,30) d:151,74 Visible
      >Windows.UI.Xaml.Controls.Border-(414x758)@(0,0) d:151,74 Visible
        >Windows.UI.Xaml.Controls.ContentPresenter-(410x754)@(2,2) d:147,70 Visible
          >Windows.UI.Xaml.Controls.Border-(410x754)@(0,0) d:147,70 Visible
            >Windows.UI.Xaml.Controls.Button-'DebugVisualButton'-(107x30)@(20,362) d:147,70 Visible
              >Windows.UI.Xaml.Controls.Grid-'RootGrid'-(107x30)@(0,0) d:107,30 Visible
                >Windows.UI.Xaml.Controls.ContentPresenter-'ContentPresenter'-(107x30)@(0,0) d:107,30 Visible
                  >Windows.UI.Xaml.Controls.ImplicitTextBlock-(87x18)@(10,6) d:87,18 Visible Text="DebugVisual"

    FlyoutBasePopupPanel::CalculateFlyoutPlacement
        desiredSize = visibleBounds.Size/*.AtMost(maxSize)*/; // 414x818
            ^ even if we ignore the maxSize the flyout is still not covering the screen
            ^ edit: because it is capped at multiple places and layers
    
    FlyoutPresenter::MeasureOverride(finalSize: 414x758) -> 151x74
    FlyoutPresenter::ArrangeOverride(finalSize: 414x758) -> 414x758

    Layouter:Arrange caps the result of FlyoutPresenter::ArrangeOverride 
        at most maxSize (414x758)...

    > it seems that in multiple places, the value is capped at the panel's max-size
    > however, in nowhere, is this max-height or max-width being assigned:
        @ src\Uno.UI\Mixins\iOS\FrameworkElementMixins.g.cs
        // whether from the property setter:
        FrameworkElement::MaxHeight.set // condition: this.GetType().Name == "FlyoutPresenter"; no-hit
        FrameworkElement::MaxHeight.get // condition: this.GetType().Name == "FlyoutPresenter"; hits
        // or, directly from the dependency property:
        public static readonly DependencyProperty MaxHeightProperty =
    > yet, this value reads 758 ????
```

Reply:
It's not too hard to figure out where the property is being assigned from, just put a breakpoint in the PropertyChanged callback. If necessary add temporary debug conditions on the control name and property type to make sure you don't get spammed by other controls.

Reply:
The MaxHeight were set by the FlyoutPresenter default style.